addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/protocol/xml/XMLConnection.java

```diff
@@ -96,53 +111,48 @@ public void send(String message) throws IOException {
-        }
+    public String sendReceive(final String message) throws IOException {
+        return postMessage(XML_GET, message, XML_END, c -> consumeResponse(c));
+    }
 
-        message = XML_GET + message + XML_END;
+    private String consumeResponse(HttpURLConnection connection) throws IOException {
+        // Read response
 
-        writeTraceFile(message);
+        Charset responseCharset = getResponseCharset(connection, StandardCharsets.UTF_8);
+        try (BufferedReader rd = new BufferedReader(new InputStreamReader(connection.getInputStream(), responseCharset))) {
+            String line;
+            StringBuilder responseBuffer = new StringBuilder();
+            while ((line = rd.readLine()) != null) {
+                responseBuffer.append(line);
+                responseBuffer.append('\r');
+            }
+            String response = responseBuffer.toString();
+            writeTraceFile(response);
+            return response;
+        }
+    }
+
+    public String getResponse(String path) throws IOException {
+        HttpURLConnection connection = null;
 
+        URL url = createBaseUrl(path);
         try {
-            URL url = createCrlUrl();
             connection = (HttpURLConnection) url.openConnection();
-            connection.setRequestMethod("POST");
-            connection.setRequestProperty("Content-Length", "" + Integer.toString(message.length()));
+            connection.setRequestMethod("GET");
 
             connection.setUseCaches(false);
             connection.setDoInput(true);
-            connection.setDoOutput(true);
+            connection.setDoOutput(false);
 
-            // Send request
-            DataOutputStream wr = new DataOutputStream(connection.getOutputStream());
-            try {
-                wr.writeBytes(message);
-                wr.flush();
-            } finally {
-                wr.close();
+            if (connection.getResponseCode() != 200) {
+                throw new IOException("Request failed");
             }
 
-            // Read response
-            BufferedReader rd = new BufferedReader(new InputStreamReader(connection.getInputStream()));
-            try {
-                String line;
-                StringBuffer responseBuffer = new StringBuffer();
-                while ((line = rd.readLine()) != null) {
-                    responseBuffer.append(line);
-                    responseBuffer.append('\r');
-                }
-                String response = responseBuffer.toString();
-
-                writeTraceFile(message);
-                return response;
-            } finally {
-                rd.close();
-            }
-        } catch (Exception e) {
-            logger.warn("post failed on: {}", message);
+            String response = consumeResponse(connection);
+            return response;
+
+        } catch (IOException e) {
+            logger.debug("Failed GET to {}", url, e);
```

Comment:
Do you catch and re-throw just to log, don't you trust the caller to log it? I would prefer to debug log the url before requesting and make sure that the caller properly handles its exceptions.